### PR TITLE
PFW-1097 M600: Fix filament change behavior when print is paused

### LINKED ISSUE #4379: [BUG] Nozzle set temperature remains at 0 during filament change when pausing print from PrusaLink
**Printer type** - MK3
**Printer firmware version** -  3.13.1

**PrusaLink** - 0.7.0
  Printing from local storage (not SD).

**Describe the bug**
  When pausing a print from the PrusaLink web interface the nozzle temperature is set to 0. Selecting to change the filament from the printer menu after that still leaves it at 0 while allowing the user to proceed with the filament change.
  
**To Reproduce**
  1. Start a print from the PrusaLink web interface from local storage, not the SD.
  2. During the print click "Pause Print" in the web interface. When that happens the nozzle temperature end will be set to 0.
  3. Go to the printer and select to change filament from the menu. The printer will remove the filament immediately and ask the user to insert new filament without raising the temperature.
  4. When the user inserts new filament the loading bar will start filling in the LCD without the extruder gear moving as the temperature by that point is too low.
  5. The printer will ask the user if the filament was extruded correctly. Answering no repeats the process where the loading bar will start filling in the LCD without the extruder gear moving.
  6. As a result of the printer being stuck in that loop it becomes impossible to manually change the temperature from the printer menu. So, the temperature needs to be set manually in PrusaLink or the user has to select that the filament was extruded correctly, immediately pause the print again, change the nozzle temperature and then repeat the process of changing the filament once again.
  
**Expected behavior**
  When the user selects to change the filament from the printer menu, the printer should set the nozzle temperature back to the print temperature for the current material before ejecting the filament and allows the user to proceed with inserting new filament.

#3148 is probably related to this issue. But since I experienced this behavior using PrusaLink rather than Octoprint I though it would be a good idea to open a new issue.

### PR DESCRIPTION
**Pull request by gudnimg**
Reimplementing PR 2390 https://github.com/prusa3d/Prusa-Firmware/pull/2390 from @wavexx ontop of latest MK3 branch with some minor additional changes

Tested on MK3S+ with MMU3.

Steps to test:
1. Start print
2. Pause print
3. Wait for while
4. Go to Tune -> Color change
5. Cold nozzle should now be re-heated
6. Filament is loaded successfully
7. Nozzle target temperature is set to 0°C again since the print is still paused.
8. Resume print => print resumes without issues

### CODE REVIEW

File: Firmware/Marlin_main.cpp [hunk 1]
```diff
@@ -3469,8 +3476,21 @@ static void gcode_M600(const bool automatic, const float x_position, const float
     
         // Feed a little of filament to stabilize pressure
         if (!automatic) {
-            current_position[E_AXIS] += FILAMENTCHANGE_RECFEED;
-            plan_buffer_line_curposXYZE(FILAMENTCHANGE_EXFEED);
+            if (isPrintPaused)
+            {
+                // Return to retracted state during a pause
+                current_position[E_AXIS] -= default_retraction;
+                plan_buffer_line_curposXYZE(FILAMENTCHANGE_RFEED);
```
Comment (gudnimg) [inline Firmware/Marlin_main.cpp:3494]: Not sure if this is actually required.

I suspect the E-move later will overwrite this:

```
       // Set E position to original
        plan_set_e_position(lastpos[E_AXIS]);
    
        memcpy(current_position, lastpos, sizeof(lastpos));
        set_destination_to_current();
```

**Comment by gudnimg** [comment]
I wonder if this fixes https://github.com/prusa3d/Prusa-Firmware/issues/4379

@3d-gussner do you have PrusaLink setup?

**Comment by gudnimg** [comment]
Rebased to sync with MK3 branch (105 commits)

**Review by 3d-gussner — APPROVED** [review]
Tested on MK404 and works as advertised.

**Pull request merged**